packages/api/db/migration/20240116175706_create_animal_tables.js

```diff
@@ -0,0 +1,189 @@
+/*
+ *  Copyright (c) 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+export const up = async function (knex) {
+  await knex.schema.createTable('animal_sex', (table) => {
+    table.increments('id').primary();
+    table.string('key').notNullable();
+  });
+
+  await knex.schema.createTable('animal_identifier_color', (table) => {
+    table.increments('id').primary();
+    table.string('key').notNullable();
+  });
+
+  await knex.schema.createTable('animal_identifier_placement', (table) => {
+    table.increments('id').primary();
+    table.integer('default_type_id').references('id').inTable('default_animal_type');
+    table.integer('custom_type_id').references('id').inTable('custom_animal_type');
+    table.string('key').notNullable();
+    table.check(
+      '?? is not null or ?? is not null',
+      ['default_type_id', 'custom_type_id'],
+      'not_null_type_id_check',
+    );
+    table.check(
+      '?? is null or ?? is null',
+      ['default_type_id', 'custom_type_id'],
+      'null_type_id_check',
+    );
+  });
+
+  await knex.schema.createTable('animal_origin', (table) => {
+    table.increments('id').primary();
+    table.string('key').notNullable();
+  });
+
+  await knex.schema.createTable('animal', (table) => {
+    table.increments('id').primary();
+    table.uuid('farm_id').notNullable().references('farm_id').inTable('farm');
+    table.integer('default_breed_id').references('id').inTable('default_animal_breed');
+    table.integer('custom_breed_id').references('id').inTable('custom_animal_breed');
+    table.integer('sex_id').references('id').inTable('animal_sex');
+    table.string('name');
+    table.date('birth_date');
```

Comment:
I am wondering if we should really avoid dates (vs datetimetz). Not necessarily because it is wrong but because we don't use them consistently in our app and the timezone info is probably relevant if we manipulate anything in the controller. 

We have some more funky functions in server.js for dealing with dates so if we go that route and keep dates - I think we would want to avoid the existing date pattern. (issue somewhat related - https://lite-farm.atlassian.net/browse/LF-3396)

Reply:
I was unsure about that too and had actually started with datetime and then changed it. It's quite possible the UI might mess up the dates because of timezones, so I'll update this